app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AdministratorControlsFragmentTest.kt

```diff
@@ -0,0 +1,725 @@
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        profileId = internalProfileId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 3)
+      verifyItemDisplayedOnAdministratorControlListItem(
+        itemPosition = 3,
+        targetView = R.id.app_information_text_view
+      )
+      verifyTextOnAdministratorListItemAtPosition(
+        itemPosition = 3,
+        targetViewId = R.id.app_version_text_view,
+        stringIdToMatch = R.string.administrator_controls_app_version
+      )
+      verifyItemDisplayedOnAdministratorControlListItem(
+        itemPosition = 4,
+        targetView = R.id.account_actions_text_view
+      )
+      verifyTextOnAdministratorListItemAtPosition(
+        itemPosition = 4,
+        targetViewId = R.id.log_out_text_view,
+        stringIdToMatch = R.string.administrator_controls_log_out
+      )
+    }
+  }
+
+  @Test
+  fun testAdministratorControlsFragment_wifiSwitchIsUncheck_autoUpdateSwitchIsUncheck() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        profileId = internalProfileId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.administrator_controls_list,
+          position = 2,
+          targetViewId = R.id.topic_update_on_wifi_switch
+        )
+      ).check(matches(Matchers.not(ViewMatchers.isChecked())))
+      scrollToPosition(position = 2)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.administrator_controls_list,
+          position = 2,
+          targetViewId = R.id.auto_update_topic_switch
+        )
+      ).check(matches(Matchers.not(ViewMatchers.isChecked())))
+    }
+  }
+
+  @Test
+  fun testAdministratorControlsFragment_clickWifiContainer_configChange_wifiSwitchIsChecked() {
+    launch<AdministratorControlsActivity>(
+      createAdministratorControlsActivityIntent(
+        profileId = internalProfileId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+      scrollToPosition(position = 2)
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.administrator_controls_list,
+          position = 2,
+          targetViewId = R.id.topic_update_on_wifi_switch
+        )
+      ).check(matches(Matchers.not(ViewMatchers.isChecked())))
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.administrator_controls_list,
+          position = 2,
+          targetViewId = R.id.auto_update_topic_switch
+        )
+      ).check(matches(Matchers.not(ViewMatchers.isChecked())))
+      onView(
+        atPositionOnView(
+          recyclerViewId = R.id.administrator_controls_list,
+          position = 2,
+          targetViewId = R.id.topic_update_on_wifi_constraint_layout
+        )
+      ).perform(click())
```

Comment:
@BenHenning @rt4914 this click isn't performed and therefore this test fails everything is correct but still it is not done.

Reply:
@rt4914 @BenHenning I have added a video below of a failing above test of the emulator.

https://user-images.githubusercontent.com/76404817/147857517-aecf3e22-c96b-4440-a332-5e2b3f7887ca.mp4

This is the code below for the above test case as we can see that we are not able to click the button it is in an infinite loop with a gap of 200ms in each iteration 
I don't know what is wrong with this code everything seems fine to me but still the outcome is very strange 
@rt4914 @BenHenning 
PTAL thanks
`
@Test
  fun testAdministratorControlsFragment_clickWifiContainer_configChange_wifiSwitchIsChecked() {
    launch<AdministratorControlsActivity>(
      createAdministratorControlsActivityIntent(
        profileId = internalProfileId
      )
    ).use {
      testCoroutineDispatchers.runCurrent()
      scrollToPosition(position = 2)
      onView(
        atPositionOnView(
          recyclerViewId = R.id.administrator_controls_list,
          position = 2,
          targetViewId = R.id.topic_update_on_wifi_switch
        )
      ).check(matches(not(isChecked())))
      onView(
        atPositionOnView(
          recyclerViewId = R.id.administrator_controls_list,
          position = 2,
          targetViewId = R.id.auto_update_topic_switch
        )
      ).check(matches(not(isChecked())))
      testCoroutineDispatchers.runCurrent()
      while (true) {
        onView(
          atPositionOnView(
            recyclerViewId = R.id.administrator_controls_list,
            position = 2,
            targetViewId = R.id.topic_update_on_wifi_constraint_layout
          )
        ).perform(click())
        Thread.sleep(200)
      }
      testCoroutineDispatchers.runCurrent()
      onView(isRoot()).perform(orientationLandscape())
      scrollToPosition(position = 2)
      onView(
        atPositionOnView(
          recyclerViewId = R.id.administrator_controls_list,
          position = 2,
          targetViewId = R.id.topic_update_on_wifi_switch
        )
      ).check(matches(isChecked()))
      onView(
        atPositionOnView(
          recyclerViewId = R.id.administrator_controls_list,
          position = 2,
          targetViewId = R.id.auto_update_topic_switch
        )
      ).check(matches(not(isChecked())))
      onView(isRoot()).perform(OrientationChangeAction.orientationPortrait())
      scrollToPosition(position = 2)
      onView(
        atPositionOnView(
          recyclerViewId = R.id.administrator_controls_list,
          position = 2,
          targetViewId = R.id.topic_update_on_wifi_switch
        )
      ).check(matches(isChecked()))
      onView(
        atPositionOnView(
          recyclerViewId = R.id.administrator_controls_list,
          position = 2,
          targetViewId = R.id.auto_update_topic_switch
        )
      ).check(matches(not(isChecked())))
    }
  }
`

Reply:
this is fixed